src/otx/core/model/entity/base.py

```diff
@@ -57,3 +60,49 @@ def forward(
             if self._customize_outputs != OTXModel._customize_outputs
             else outputs
         )
+
+    def register_load_state_dict_pre_hook(self, model_classes: list[str], ckpt_classes: list[str]) -> None:
+        """Register load_state_dict_pre_hook.
+
+        Args:
+            model_classes (list[str]): Class names from training data.
+            ckpt_classes (list[str]): Class names from checkpoint state dictionary.
+        """
+        self.model_classes = model_classes
+        self.ckpt_classes = ckpt_classes
+        self._register_load_state_dict_pre_hook(self.load_state_dict_pre_hook)
+
+    def load_state_dict_pre_hook(self, state_dict: dict[str, torch.Tensor], prefix: str, *args, **kwargs) -> None:
+        """Modify input state_dict according to class name matching before weight loading."""
+        model2ckpt = self.map_class_names(self.model_classes, self.ckpt_classes)
+
+        for param_name in self.state_dict():
+            model_param = self.state_dict()[param_name].clone()
+            ckpt_param = state_dict[prefix + param_name]
+            if model_param.shape != ckpt_param.shape:
```

Comment:
How can we cover the following case?

```python
          src_classes = ['person', 'car', 'tree']
          dst_classes = ['sky', 'ball', 'person']
```

Reply:
I believe that this can be fundamentally solved by implementing a logic to search the prediction layer. Like

> p.s. Is it fundamentally possible to solve the following problem by implementing own _load_state_dict_incrementally() for each OTXModel? We can have some general solutions at some model classes, but it is not possible to infer what is the head layer of the model by machine automatically.

Reply:
I believe that `meta_info` decides the shape of `OTXModel`, so that `meta_info` should be a member of `OTXModel`. In addition, it means that `OTXModel.load_state_dict()` should be able to handle the increamental learning case by itself.

Reply:
Done by 4517f09 and 486a2e3
I added auto detection of classification layers by comparing two models with different num_classes